daphne/src/error/aborts.rs

```diff
@@ -149,15 +149,26 @@ impl DapAbort {
 
     /// Abort due to unexpected value for HTTP content-type header.
     pub fn content_type<S>(req: &DapRequest<S>, expected: DapMediaType) -> Self {
-        let want_str = expected.as_str_for_version(req.version);
-
-        if let Some(got_str) = req.media_type.map(|mt| mt.as_str_for_version(req.version)) {
-            Self::BadRequest(format!(
-                "unexpected content-type: got {got_str}; want {want_str}"
-            ))
-        } else {
-            Self::BadRequest(format!("missing content-type: expected {want_str}"))
-        }
+        let Some(want_content_type) = expected.as_str_for_version(req.version) else {
+            return Self::BadRequest(format!(
+                "unexpected content-type for DAP version {:?}",
+                req.version
+            ));
```

Comment:
If this else branch happens, doesn't that mean that we expected a media type that's unsupported by the version? Shouldn't this be a 500 instead? If we expect a media type that this version could never have supported than this was a mistake by our code

Reply:
Yes, 500 is more appropriate, however since the return type is `DapAbort`, we can't force this here. I also don't want to add another variant of `DapAbort`, since we recently moved from handling 500s via `DapError`.

Another option would be to just panic, as we're already doing. Would you prefer that?

Reply:
I'd prefer a panic (which will be turned into a 500 anyway) than this. Panics are okay for code that should *never* happen. And we don't expect this to ever happen.